gap_nrf528xx-advertisement.go

```diff
@@ -84,3 +84,9 @@ func (a *Advertisement) Stop() error {
 	errCode := C.sd_ble_gap_adv_stop(a.handle)
 	return makeError(errCode)
 }
+
+// SetServiceData sets the service data for the advertisement.
+func (a *Advertisement) SetServiceData(sd []ServiceDataElement) error {
+	// TODO: implement
+	return nil
```

Comment:
Looks like this can be implemented here, but it needs a different API.
https://docs.nordicsemi.com/bundle/s132_v6.0.0_api/page/group_b_l_e_g_a_p_f_u_n_c_t_i_o_n_s.html#ga9969047f4e7485c3f856c841978cc31a
Instead of a `SetServiceData`, you'd need to construct an entirely new advertisement payload (`bluetooth.AdvertisementOptions`) and set it as a whole.